docs/addons/addons.md

```diff
@@ -0,0 +1,148 @@
+# Addon Submission Pipeline
+## Terminology:
+
+* listed - Addon is viewable on AMO by end-users. 
+* unlisted - Addon is available only to the addon uploader for download, and servable
+  outside of AMO.
+* Language Pack - A translation pack for Gecko strings bundled as an addon.
+
+## Background
+
+Langauge packs now need to be signed (as of Gecko 60.1esr/61.0) to be run inside of
+Firefox. In order to do that our release process submits every language pack to AMO
+for signing.
+
+### Restrictions
+
+In order to achieve this we have to contend with a few current restrictions of AMO
+itself, as well as Firefox the product, and correlate them with our release process.
+
+* Version number of the language pack is baked into the language pack itself.
+* AMO only accepts one copy of each language pack for each specific version (a
+  new upload needs a new version)
+* AMO does not support "promoting" an upload to `listed` if it was previously
+  `unlisted`
+* AMO needs the first `listed` version of an addon to be submitted via the admin web
+  interface manually, the first time.
+
+### Process
+
+* Firefox, for every code checkin, builds an English (US) copy that is suitable for
+  shipping to end users, as part of this build we generate the en-US language pack.
+* Once Release Management indicates we want to build a release, we kickoff the 'promote'
+  phase of the release process, which starts the Localized Repackages, which generate
+  each langauges language pack.
+* The language packs get sent off to AMO on an addon submission task, this task
+  submits and retries while it waits for AMO to sign the langpacks. Timing
+  out (with a signal to retry the task) if needed.
+* The Addon Submission task is idempotent, so if a given langpack version is uploaded
+  it will retrieve the same upload on subsequent calls (and resubmit any uploads
+  which failed)
+* The Addon Submission task exposes the signed langpacks as task artifacts.
+* Beetmover moves the language packs to the release folder, allowing them to be
+  easily downloaded by users not on AMO.
+
+## How To...
+### How to handle New Languages for Release
+
+* Once the promote phase is kicked off, any new language packs will fail the addon
```

Comment:
I forgot, @Callek - there is no way to avoid fixing new locales prior to an actual release yes? If there is a way and it should be done by $someone, should we mention that here so it's clear why we need to do this during a release (hot fix)?

Reply:
Right now, :flod has a human item to notify me when a new locale goes in. We could also setup Herald rules for the central files that are involved to notify us.

My current design/plan is to set a rw future task for the work when $release goes to RC... -- Would be great to avoid these extra human steps with AMO changes though...